R_functions/process_estimates_pe.R

```diff
@@ -0,0 +1,186 @@
+             !is.nan(est) & !is.nan(var) & !is.nan(l95) & !is.nan(u95))
+  
+  rows_post <- nrow(filtered_data)
+  n_nan <- rows_pre - rows_post
+  
+  #message to user with red text
+  message(paste("NaN values present in the PE effort estimates.", n_nan, "rows filtered out for upload."))
+  # cat("\033[31m", "NaN values present in the PE effort estimates.", n_nan, "rows filtered out for upload.", "\033[39m")
+  cat("\n")
+  
+  #transformation
+  transformed_pe_data$pe_summarized_effort <- transformed_pe_data$pe_summarized_effort |>
+    group_by(analysis_id, project_name, fishery_name, model_type) |>
+    summarise(est_sum = sum(est), .groups = "keep") |> 
+    pivot_longer(cols = c(est_sum),
+                 names_to = "estimate_type",
+                 values_to = "value") |> 
+    #set min date as start of monitoring period
+    mutate(min_event_date = as.Date(params$est_date_start),
+           #set max_event_date as sys.date if in-season, set as est_end_date if out of monitoring period 
+           max_event_date = as.Date(
+             ifelse(
+               Sys.Date() <= params$est_date_end, Sys.Date(),
+               params$est_date_end
+             ))
+    )
+  
+  #tidy output
+  transformed_pe_data$pe_summarized_effort <- transformed_pe_data$pe_summarized_effort |>
+    ungroup() |>
+    mutate(estimate_category = "effort") |>
+    relocate("estimate_category", .after = "model_type")
+
+  # # Effort transformation - !!! ORIGINAL
+  # transformed_pe_data$pe_summarized_effort <- transformed_pe_data$pe_summarized_effort |>
+  #   group_by(analysis_id, project_name, fishery_name, model_type) |>
+  #   summarise(est_sum = sum(est)
+  #   ) |>
+  #   pivot_longer(cols = c(est_sum),
+  #                names_to = "estimate_type",
+  #                values_to = "value") |> 
+  #   #set min date as start of monitoring period
+  #   mutate(min_event_date = as.Date(params$est_date_start),
+  #   #set max_event_date as sys.date if in-season, set as est_end_date if out of monitoring period 
+  #          max_event_date = as.Date(
+  #            ifelse(
+  #              Sys.Date() <= params$est_date_end, Sys.Date(),
+  #              params$est_date_end
+  #              ))
+  #   )
+  # 
+  # #tidy output
+  # transformed_pe_data$pe_summarized_effort <- transformed_pe_data$pe_summarized_effort |>
+  #   drop_na() |>
+  #   ungroup() |>
+  #   mutate(estimate_category = "effort") |>
+  #   relocate("estimate_category", .after = "model_type")
+
+  ### !!! end of effort transformation ####
+    
+  # Catch transformation
+  transformed_pe_data$pe_summarized_catch <- transformed_pe_data$pe_summarized_catch |>
+    group_by(analysis_id, project_name, fishery_name, model_type, est_cg) |> #includes catch group col
+    summarise(n_obs_sum = sum(n_obs),
+              N_days_open_sum = sum(N_days_open),
+              est_sum = {
+                #error handling for NA values in the estimate column
```

Comment:
If I'm remembering correctly, this particular NaN issue could only arise in the effort data and is based on 0/0 rather than a missing data value. Within a given stratum, if there is catch data but no effort to expand by, or vice versa, R introduces either a zero or a NaN value when calculating catch_est_mean or ang_hrs_mean, respectively. 

https://github.com/wdfw-fp/CreelEstimates/blob/d8ee13275e049646e898c339192e88179eef040d/R_functions/est_pe_effort.R#L41-L51

For example, when mean is calculated for catch and there is zero catch but effort (0/1 = 0) and when there is catch but zero effort (1/0 = Inf). However, NaN values are introduced when there is zero catch and zero effort within a given stratum (0/0 = NaN).
![image](https://github.com/wdfw-fp/CreelEstimates/assets/62571975/b99f6dfd-f9ed-4e93-bd3c-0adc1a562d4d)

I think we were going to wait until after the ETL process was introduced to address that issue in est_pe_catch and est_pe_effort, which is why I have the NaN values filtered out for now with a message to the user that the issue is occurring. As for the Inf values, either those are being forced to zeros already by summarize() etc, or we haven't yet encountered that situation.